program/freedom-metal-program.wake

```diff
@@ -43,44 +43,46 @@ tuple FreedomBSP =
   global PlatformHeader: Path
   global Attributes:     FreedomMakeAttributes
 
-global def makeFreedomBSP coreDTS customOverlay type prefix =
+global def makeFreedomBSP coreDTS customOverlay type prefix machineName =
```

Comment:
Wait, this makes a backwards-incompatible change to this API. Since it's `global`, let me quickly check if we have any other codebases that reference this.

Reply:
OK, this seems to not be used anywhere outside of this file. Is this something we should consider making non-global? If not, then I think we'll want to more gracefully handle forwards-compatibility by wrapping the arguments into `tuples`, for which we can define accessor methods for new fields in an incremental fashion.